Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -68,6 +68,8 @@ public class RepositoryPackageContainer: PackageContainer, CustomStringConvertib
 
     private var knownVersionsCache = ThreadSafeBox<[Version: String]>()
     private var reversedVersionsCache = ThreadSafeBox<[Version]>()
+    private var manifestsCache = ThreadSafeDictionary<Revision, Manifest>()
```

Comment:
My only concern here would be about the size of this cache, but it's almost certainly worth it for the performance wins.  Based on my understanding of this logic, it doesn't look as if any of these entries would become completely irrelevant — any re-resolution would presumably revisit the same revisions again.